sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -669,7 +670,11 @@ public static Object convertAvroFieldStrict(
                 .fromBytes(byteBuffer.duplicate(), type.type, logicalType);
         return convertDecimal(bigDecimal, fieldType);
       } else if (logicalType instanceof LogicalTypes.TimestampMillis) {
-        return convertDateTimeStrict((Long) value, fieldType);
+        if (value instanceof DateTime) {
+          return convertDateTimeStrict(((DateTime) value).getMillis(), fieldType);
```

Comment:
As the TimestampConversion is set in a static initialization block, the expectation of the initial code is to covert long to JodaTime during deserialization.  So with that expectation the else part will not be needed in this case.

Also how are the other logical-types of Avro handled when converting to Row ?

Reply:
To make this slightly more general, can we instead check for instanceof ReadableInstant (which is a base interface for DateTime).